tests/unit-tests/order/crud.php

```diff
@@ -7,6 +7,26 @@
 class WC_Tests_CRUD_Orders extends WC_Unit_Test_Case {
 
 	/**
+	 * @todo move this to the bottom of this class.
+	 */
+	function test_remove_coupon() {
+		$order = WC_Helper_Order::create_order();
+		$order->calculate_totals( true );
+
+		$coupon = new WC_Coupon;
+		$coupon->set_code( 'test' );
+		$coupon->set_discount_type( 'fixed_cart' );
+		$coupon->set_amount( 10 );
+		$coupon->save();
+
+		$order->add_discount( 'test' );
+		$this->assertEquals( 40, $order->get_total() );
+
```

Comment:
This test fails because the order total is $20 here. That seems very off, or I'm doing something wrong.

($10 * 4 products) + $10 shipping = $50 before discounts.
$50 - $10 cart discount = $40
Right?